src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -247,8 +247,14 @@ private void addShutdownListenerToChannel(Channel channel, Boolean withRecovery)
                         errorBuilder.append(channelCause);
                         String errorString = errorBuilder.toString();
                         LOGGER.warn(errorString);
-                        if (withRecovery && errorString.contains("PRECONDITION_FAILED")) {
-                            recoverSubscriptionsOfChannel(channel.getChannelNumber());
+                        if (withRecovery &&
+                                (errorString.contains("reply-text=PRECONDITION_FAILED")
```

Comment:
I think we should move these error texts into a static list and create a method to check if the error is recoverable by iterating over this list and checking if error text contains expected substrings